src/wiki/views/article.py

```diff
@@ -505,31 +510,35 @@ def form_valid(self, form):
 
             root_len = len(descendants[0].path)
 
+            for x in descendants:
+                # Without this descendant.get_ancestors() and as a result descendant.path
+                # is wrong after the first create_article() due to path caching
+                str(x.path)
+
             for descendant in descendants:
                 # Without this descendant.get_ancestors() and as a result descendant.path
                 # is wrong after the first create_article() due to path caching
-                descendant.refresh_from_db()
+                # descendant.refresh_from_db()
                 dst_path = descendant.path
```

Comment:
Any reason for going back to str(descendants), which this effectively is?

It works only because of the current caching implementation in URLPath.
for: str(x.path) prefills all URLPath.cached_ancestors, so that later, after
the calls to create_urlpath(), get_ancestors() is not called anymore from
URLPath.path(). The underlying problem, completely wrong get_ancestors()
because of a messed up tree, still exists.

That is perhaps a problem of how create_urlpath() uses mptt, or, probably
more likely, a bug/missing feature in mptt itself.

If that was planed I can merge it as well as is.